File: src/main/java/com/hsmchurch/app/video/application/VideoSupportService.java
package com.hsmchurch.app.video.application;

import com.hsmchurch.app.reply.application.ReplyService;
import com.hsmchurch.app.video.domain.LikeTag;
import com.hsmchurch.app.video.ui.response.LikeUser;
import com.hsmchurch.app.video.ui.response.ReplyForVideo;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

import static java.util.stream.Collectors.toList;

@Service
@RequiredArgsConstructor
public class VideoSupportService {

    private final LikeTagService likeTagService;
    private final ReplyService replyService;

    public List<Long> likeList(final Long accountId) {
        return likeTagService.findAllByAccountId(accountId).stream()
                .map(LikeTag::getVideoId)
                .collect(toList());
    }

    public List<LikeUser> likeUsers(final Long videoId) {
        return likeTagService.findAllByVideo(videoId);
    }

    public List<ReplyForVideo> replies(final Long id) {
        return replyService.findAllByVideo2(id);
    }
}

File: src/main/java/com/hsmchurch/app/video/application/VideoSupportService.kt
package com.hsmchurch.app.video.application

import com.hsmchurch.app.reply.application.ReplyService
import com.hsmchurch.app.video.ui.response.LikeUser
import com.hsmchurch.app.video.ui.response.ReplyForVideo
import lombok.RequiredArgsConstructor
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class VideoSupportService constructor(
        private val likeTagService: LikeTagService,
        private val replyService: ReplyService
){

    fun likeList(accountId: Long,
                 pageable: Pageable): Page<Long> = likeTagService.findAllByAccountId(accountId, pageable).map { it.videoId }

    fun likeUsers(videoId: Long): List<LikeUser> = likeTagService.findAllByVideo(videoId)

    fun replies(id: Long): List<ReplyForVideo> = replyService.findAllByVideo(id)
}
